packages/runtime/container-runtime/src/summary/documentSchema.ts

```diff
@@ -461,24 +498,53 @@ export class DocumentsSchemaController {
 	 * @param documentMetadataSchema - current document's schema, if present.
 	 * @param features - features of the document schema that current session wants to see enabled.
 	 * @param onSchemaChange - callback that is called whenever schema is changed (not called on creation / load, only when processing document schema change ops)
+	 * @param minVersionForCollab - minimum version of the runtime that is supported for collaboration with this document.
 	 */
 	constructor(
 		existing: boolean,
 		snapshotSequenceNumber: number,
 		documentMetadataSchema: IDocumentSchema | undefined,
 		features: IDocumentSchemaFeatures,
 		private readonly onSchemaChange: (schema: IDocumentSchemaCurrent) => void,
+		minVersionForCollab: MinimumVersionForCollab,
+		logger: ITelemetryLoggerExt,
 	) {
 		// For simplicity, let's only support new schema features for explicit schema control mode
 		assert(
 			features.disallowedVersions.length === 0 || features.explicitSchemaControl,
 			0x949 /* not supported */,
 		);
 
+		// We check the document's metadata to see if there is a minVersionForCollab. If it's not an existing document or
+		// if the document is older, then it won't have one. If it does have a minVersionForCollab, we check if it's greater
+		// than this client's runtime version. If so, we log a telemetry event to warn the customer that the client is outdated.
+		// Note: We only send a warning because we already found that this client **can** understand the existing document's
+		// schema (the `DocumentsSchemaController` constructor throws otherwise). However, we still want to issue a warning to
+		// the customer since it may be a sign that the customer is not properly waiting for saturation before updating their
+		// `minVersionForCollab` value, which could cause disruptions to users in the future.
+		const existingMinVersionForCollab = documentMetadataSchema?.minVersionForCollab;
+		if (
+			existingMinVersionForCollab !== undefined &&
+			gt(existingMinVersionForCollab, pkgVersion)
+		) {
+			const warnMsg = `WARNING: The version of Fluid Framework used by this client (${pkgVersion}) is not supported by this document! Please upgrade to version ${existingMinVersionForCollab} or later to ensure compatibility.`;
+			logger.sendTelemetryEvent({
+				eventName: "ContainerRuntime:MinVersionForCollabWarning",
```

Comment:
Prefix would be redundant with the namespace that's already been built up on the logger instance.
```suggestion
				eventName: "MinVersionForCollabWarning",
```